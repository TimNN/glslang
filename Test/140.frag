#version 140

varying vec4 v;

in vec4 i;
out vec4 o;

void main()
{
}
