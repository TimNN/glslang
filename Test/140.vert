#version 140

uniform isamplerBuffer sbuf;

layout(std140) uniform blockName {
    int anonMem;
};

void main()
{
    int id = gl_InstanceID;
    id += anonMem;
    id += texelFetch(sbuf, 8).w;
    gl_ClipVertex;      // could be ERROR, but compiling under compatibility profile
    gl_Color;           // could be ERROR, but compiling under compatibility profile
    gl_LightSource[0];  // could be ERROR, but compiling under compatibility profile
    gl_DepthRange.far;
    gl_TexCoord;        // could be ERROR, but compiling under compatibility profile
    gl_FogFragCoord;    // could be ERROR, but compiling under compatibility profile
    gl_FrontColor;      // could be ERROR, but compiling under compatibility profile
}

out vec4 gl_Position;  // ERROR

layout(location = 9) in vec4 locBad;  // ERROR

#extension GL_ARB_explicit_attrib_location : enable

layout(location = 9) in vec4 loc;

#extension GL_ARB_separate_shader_objects : enable

out vec4 gl_Position;
in vec4 gl_Position;   // ERROR
out vec3 gl_Position;  // ERROR

out float gl_PointSize;
out vec4 gl_ClipVertex;
out float gl_FogFragCoord;
