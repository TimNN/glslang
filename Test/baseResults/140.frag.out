140.frag
WARNING: 0:3: varying deprecated in version 130; may be removed in future release
ERROR: 0:17: '#error' : GL_ES is not set  
ERROR: 0:20: 'fragment-shader struct input' : not supported for this version or the enabled extensions 
ERROR: 0:24: 'location' : not supported for this version or the enabled extensions 
ERROR: 0:24: 'location qualifier on input' : not supported for this version or the enabled extensions 
ERROR: 0:32: 'assign' :  l-value required "v" (can't modify shader input)
ERROR: 0:32: 'out' : Non-L-value cannot be passed for 'out' or 'inout' parameters. 
ERROR: 6 compilation errors.  No code generated.


Shader version: 140
Requested GL_ARB_separate_shader_objects
ERROR: node is still EOpNull!
0:10  Function Definition: main( (void)
0:10    Function Parameters: 
0:12    Sequence
0:12      Sequence
0:12        move second child to first child (float)
0:12          'clip' (float)
0:12          direct index (smooth float)
0:12            'gl_ClipDistance' (smooth in 5-element array of float)
0:12            Constant:
0:12              2 (const int)
0:22  Sequence
0:22    move second child to first child (float)
0:22      'patch' (float)
0:22      Constant:
0:22        3.100000
0:30  Function Definition: foo( (void)
0:30    Function Parameters: 
0:32    Sequence
0:32      Sequence
0:32        move second child to first child (2-component vector of float)
0:32          'r1' (2-component vector of float)
0:32          modf (2-component vector of float)
0:32            vector swizzle (2-component vector of float)
0:32              'v' (smooth in 4-component vector of float)
0:32              Sequence
0:32                Constant:
0:32                  0 (const int)
0:32                Constant:
0:32                  1 (const int)
0:32            vector swizzle (2-component vector of float)
0:32              'v' (smooth in 4-component vector of float)
0:32              Sequence
0:32                Constant:
0:32                  2 (const int)
0:32                Constant:
0:32                  3 (const int)
0:33      Sequence
0:33        move second child to first child (2-component vector of float)
0:33          'r2' (2-component vector of float)
0:33          modf (2-component vector of float)
0:33            vector swizzle (2-component vector of float)
0:33              'o' (out 4-component vector of float)
0:33              Sequence
0:33                Constant:
0:33                  0 (const int)
0:33                Constant:
0:33                  1 (const int)
0:33            vector swizzle (2-component vector of float)
0:33              'o' (out 4-component vector of float)
0:33              Sequence
0:33                Constant:
0:33                  2 (const int)
0:33                Constant:
0:33                  3 (const int)
0:?   Linker Objects
0:?     'v' (smooth in 4-component vector of float)
0:?     'i' (smooth in 4-component vector of float)
0:?     'o' (out 4-component vector of float)
0:?     'gl_ClipDistance' (smooth in 5-element array of float)
0:?     'gl_ClipDistance' (smooth in 5-element array of float)
0:?     's' (smooth in structure{float f})
0:?     'patch' (float)
0:?     'vl' (layout(location=3 ) smooth in 4-component vector of float)
0:?     'vl2' (layout(location=4 ) smooth in 4-component vector of float)


Linked fragment stage:


Shader version: 140
Requested GL_ARB_separate_shader_objects
ERROR: node is still EOpNull!
0:10  Function Definition: main( (void)
0:10    Function Parameters: 
0:12    Sequence
0:12      Sequence
0:12        move second child to first child (float)
0:12          'clip' (float)
0:12          direct index (smooth float)
0:12            'gl_ClipDistance' (smooth in 5-element array of float)
0:12            Constant:
0:12              2 (const int)
0:22  Sequence
0:22    move second child to first child (float)
0:22      'patch' (float)
0:22      Constant:
0:22        3.100000
0:30  Function Definition: foo( (void)
0:30    Function Parameters: 
0:32    Sequence
0:32      Sequence
0:32        move second child to first child (2-component vector of float)
0:32          'r1' (2-component vector of float)
0:32          modf (2-component vector of float)
0:32            vector swizzle (2-component vector of float)
0:32              'v' (smooth in 4-component vector of float)
0:32              Sequence
0:32                Constant:
0:32                  0 (const int)
0:32                Constant:
0:32                  1 (const int)
0:32            vector swizzle (2-component vector of float)
0:32              'v' (smooth in 4-component vector of float)
0:32              Sequence
0:32                Constant:
0:32                  2 (const int)
0:32                Constant:
0:32                  3 (const int)
0:33      Sequence
0:33        move second child to first child (2-component vector of float)
0:33          'r2' (2-component vector of float)
0:33          modf (2-component vector of float)
0:33            vector swizzle (2-component vector of float)
0:33              'o' (out 4-component vector of float)
0:33              Sequence
0:33                Constant:
0:33                  0 (const int)
0:33                Constant:
0:33                  1 (const int)
0:33            vector swizzle (2-component vector of float)
0:33              'o' (out 4-component vector of float)
0:33              Sequence
0:33                Constant:
0:33                  2 (const int)
0:33                Constant:
0:33                  3 (const int)
0:?   Linker Objects
0:?     'v' (smooth in 4-component vector of float)
0:?     'i' (smooth in 4-component vector of float)
0:?     'o' (out 4-component vector of float)
0:?     'gl_ClipDistance' (smooth in 5-element array of float)
0:?     'gl_ClipDistance' (smooth in 5-element array of float)
0:?     's' (smooth in structure{float f})
0:?     'patch' (float)
0:?     'vl' (layout(location=3 ) smooth in 4-component vector of float)
0:?     'vl2' (layout(location=4 ) smooth in 4-component vector of float)

